zipkin-storage/elasticsearch-http/src/test/java/zipkin/storage/elasticsearch/http/ElasticsearchHttpSpanConsumerTest.java

```diff
@@ -110,83 +110,60 @@ public void writesSpanNaturallyWhenNoTimestamp() throws Exception {
     es.enqueue(new MockResponse());
 
-    String name = "{\"foo\":\"bar\"}";
-    String nameEscaped = "{\\\"foo\\\":\\\"bar\\\"}";
-
-    accept(TestObjects.TRACE.get(0).toBuilder().name(name).build());
+    accept(TestObjects.TRACE.get(0));
 
-    assertThat(es.takeRequest().getBody().readByteString().utf8()).endsWith(
-      "\"_type\":\"servicespan\",\"_id\":\"web|" + nameEscaped + "\"}}\n"
-        + "{\"serviceName\":\"web\",\"spanName\":\"" + nameEscaped + "\"}\n"
-    );
+    RecordedRequest request = es.takeRequest();
+    assertThat(request.getPath())
+      .isEqualTo("/_bulk?pipeline=zipkin");
   }
 
   @Test
-  public void traceIsSearchableBySRServiceName() throws Exception {
+  public void indexesServiceSpan_basedOnAnnotationTimestamp() throws Exception {
     es.enqueue(new MockResponse());
 
-    Span clientSpan = Span.builder().traceId(20L).id(22L).name("").parentId(21L).timestamp(0L)
-        .addAnnotation(Annotation.create(0, CLIENT_SEND, TestObjects.WEB_ENDPOINT))
-        .build();
-
-    Span serverSpan = Span.builder().traceId(20L).id(22L).name("get").parentId(21L)
-        .addAnnotation(Annotation.create(1000, SERVER_RECV, TestObjects.APP_ENDPOINT))
-        .build();
-
-    accept(serverSpan, clientSpan);
+    Annotation foo = Annotation.create(
+      TimeUnit.DAYS.toMicros(365), // 1971-01-01
+      "foo",
+      TestObjects.APP_ENDPOINT
+    );
 
-    // make sure that both timestamps are in the index
-    assertThat(es.takeRequest().getBody().readByteString().utf8())
-        .contains("{\"timestamp_millis\":1")
-        .contains("{\"timestamp_millis\":0");
-  }
+    Span span = Span.builder().traceId(1L).id(2L).parentId(1L).name("s").addAnnotation(foo).build();
 
-  @Test
-  public void indexesServiceSpan_multipleServices() throws Exception {
-    es.enqueue(new MockResponse());
+    // sanity check data
+    assertThat(span.timestamp).isNull();
+    assertThat(guessTimestamp(span)).isNull();
 
-    Span span = TestObjects.TRACE.get(1);
     accept(span);
 
-    assertThat(es.takeRequest().getBody().readByteString().utf8())
-        .contains(
-            "\"_type\":\"servicespan\",\"_id\":\"app|get\"}}\n"
-                + "{\"serviceName\":\"app\",\"spanName\":\"get\"}\n"
-        )
-        .contains(
-            "\"_type\":\"servicespan\",\"_id\":\"web|get\"}}\n"
-                + "{\"serviceName\":\"web\",\"spanName\":\"get\"}\n"
-        );
+    // index timestamp is the server timestamp, not current time!
+    assertThat(es.takeRequest().getBody().readByteString().utf8()).contains(
+      "{\"index\":{\"_index\":\"zipkin-1971-01-01\",\"_type\":\"span\"}}\n",
+      "{\"index\":{\"_index\":\"zipkin-1971-01-01\",\"_type\":\"servicespan\",\"_id\":\"app|s\"}}\n"
+    );
   }
 
   @Test
   public void indexesServiceSpan_basedOnGuessTimestamp() throws Exception {
     es.enqueue(new MockResponse());
 
     Annotation cs = Annotation.create(
-        TimeUnit.DAYS.toMicros(365), // 1971-01-01
```

Comment:
please don't change the behavior of existing tests (if that's what's going on here.. can't tell if this is changing the test or if the file has been re-ordered). Main thing we need to know is if this is changing behavior or not.. usually we add tests to show new things happening and keeping the old tests help us know we didn't break anything.